File: bloaters/long-method/replace-temp-with-query/C#/Problem.cs
using System;

namespace CSharp
{
    public class Problem
    {
        public int Quantity { get; set; }
        public double ItemPrice { get; set; }

        public Problem()
        {
        }

        public void PrintTotal()
        {
            Console.WriteLine($"Total: {CalculateTotal()}");
        }
        
        private double CalculateTotal()
        {
            // Calculate Base Price
            double basePrice = Quantity * ItemPrice;

            if (basePrice > 1000)
                return basePrice * 0.95;
            else
                return basePrice * 0.98;
        }
    }
}

File: bloaters/long-method/replace-temp-with-query/Typescript/problem.ts
export class Problem {
  public quantity!: number;
  public itemPrice!: number;

  constructor() {}

  public printTotal(): void {
    console.log("Total: ", this.calculateTotal());
  }

  private calculateTotal(): number {
    // Calculate Base Price
    let basePrice = this.quantity * this.itemPrice;

    if (basePrice > 1000) return basePrice * 0.95;
    else return basePrice * 0.98;
  }
}
